pynest/nest/lib/hl_api_sonata.py

```diff
@@ -0,0 +1,693 @@
+
+    def _create_edges_maps(self):
+        """Create a collection of maps of edge properties.
+
+        Creates a map between edge type id and edge (synapse) properties for
+        each edge CSV file. The associated edge HDF5 filename is included in
+        the map as well.
+        """
+
+        # Iterate edge config files
+        for edges_conf in self._conf["networks"]["edges"]:
+            edges_map = {}
+            edges_csv_fn = edges_conf["edge_types_file"]
+            edges_df = pd.read_csv(edges_csv_fn, sep=r"\s+")
+
+            if "model_template" not in edges_df.columns:
+                msg = (
+                    "Missing the required 'model_template' header "
+                    f"specifying NEST synapse models in {edges_csv_fn}."
+                )
+                raise ValueError(msg)
+
+            # Rename column labels to names used by NEST. Note that rename
+            # don't throw an error for extra labels (we want this behavior)
+            edges_df.rename(
+                columns={"model_template": "synapse_model", "syn_weight": "weight"},
+                inplace=True,
+            )
+
+            models_arr = edges_df["synapse_model"].to_numpy()
+            is_one_model = (models_arr[0] == models_arr).all()
+            have_dynamics = "dynamics_params" in edges_df.columns
+            edges_df_cols = set(edges_df.columns)
+
+            if is_one_model:
+                synapse_model = models_arr[0]
+                # Set of settable parameters
+                settable_params = set([*GetDefaults(synapse_model)])
+                # Parameters to extract (elements common to both sets)
+                extract_cols = list(settable_params & edges_df_cols)
+                if have_dynamics:
+                    extract_cols.append("dynamics_params")
+
+                syn_specs = edges_df.set_index("edge_type_id")[extract_cols].to_dict(
+                    orient="index"
+                )
+
+                if have_dynamics:
+                    # Include parameters from JSON file in the map
+                    for edge_type_id, syn_spec in syn_specs.copy().items():
+                        params_path = PurePath(
+                            self._conf["components"]["synaptic_models_dir"],
+                            syn_spec["dynamics_params"],
+                        )
+                        with open(params_path) as fp:
+                            params = json.load(fp)
+
+                        syn_specs[edge_type_id].update(params)
+                        syn_specs[edge_type_id].pop("dynamics_params")
+            else:
+                # More than one synapse model in CSV file
+                syn_specs = {}
+                idx_map = {k: i for i, k in enumerate(list(edges_df), start=1)}
+
+                for row in edges_df.itertuples(name=None):
+                    # Set of settable parameters
+                    settable_params = set([*GetDefaults(row[idx_map["synapse_model"]])])
```

Comment:
The code following here looks very similar to the code in the `if is_one_model` case. The entire method here is also very long. Could you extract the code in the `if` and (most of) the `else` clauses into separate method(s) and maybe even avoid code duplication?